test/bdd/pkg/route/route_sdk_steps.go

```diff
@@ -33,13 +34,48 @@ func (d *SDKSteps) CreateRouteClient(agentID string) error {
 		return fmt.Errorf("failed to create new route client: %w", err)
 	}
 
+	events := make(chan service.DIDCommAction)
+
+	err = routeClient.RegisterActionEvent(events)
+	if err != nil {
+		return fmt.Errorf("failed to register %s for action events on their routing client : %w", agentID, err)
+	}
+
+	callbacks := make(chan interface{})
+
 	d.bddContext.RouteClients[agentID] = routeClient
+	d.bddContext.RouteCallbacks[agentID] = callbacks
+
+	go d.handleEvents(events, callbacks)
 
 	return nil
 }
 
+func (d *SDKSteps) handleEvents(events chan service.DIDCommAction, callbacks chan interface{}) {
+	for event := range events {
+		logger.Debugf("handling event: %+v", event)
+
+		c := <-callbacks
+
+		logger.Debugf("received callback: %+v", c)
+		event.Continue(c)
+	}
+}
+
+// ApproveRequest approves a routing protocol request with the given args.
+func (d *SDKSteps) ApproveRequest(agentID string, args interface{}) {
+	c, found := d.bddContext.RouteCallbacks[agentID]
+	if !found {
+		logger.Warnf("no callback channel found for %s", agentID)
+	}
+
+	c <- args
```

Comment:
It is a `deadlock`.  When `!found` condition is true, `c` will be a `nil`. Sending value to a nil channel blocks forever.

Reply:
done